In [1]:

import vcf
import gzip
import subprocess
import pandas as pd
#from jupyterthemes  import jtplot

#jtplot.style(theme="onedork")


/home/felipe/anaconda3/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: Warning:
  warnings.warn(x, RRuntimeWarning)
/home/felipe/anaconda3/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning:  namespace ‘SparkR’ is not available and has been replaced
by .GlobalEnv when processing object ‘teenagers’

  warnings.warn(x, RRuntimeWarning)
/home/felipe/anaconda3/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning:  namespace ‘twitteR’ is not available and has been replaced
by .GlobalEnv when processing object ‘tweets’

  warnings.warn(x, RRuntimeWarning)


In [115]:

vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123.vcf.gz',compressed=True,encoding='UTF-8')


gene_id_presence=[]
gene_id_name=[]
for record in vcf_reader:
    pres=set(record.INFO.keys()).intersection(set(['GENEINFO']))
    gene_id_presence.append(pres)
    if pres!= set():
        gene_id_name.append(pres)
    else:
        gene_id_name.append(" ")


    

Calculo a cuantos tendria que mapear por coordenadas 

In [118]:
gene_id_presence.count(set())

137

In [167]:
gene_id_presence.index(set())

14622

In [7]:
cnt=0

vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123.vcf.gz',compressed=True,encoding='UTF-8')

for variant in vcf_reader:
    cnt=cnt+1
    if(cnt==14622):
        esta=variant
        break
        
        


str(esta.CHROM)+"-"+str(esta.POS)+"-"+str(esta.REF)+"-"+str(esta.ALT)



'1-104094352-C-[T]'

In [8]:
esta.ID

'587368'

In [9]:
esta.INFO

{'ALLELEID': 578325,
 'CLNDISDB': ['MedGen:CN257749', 'OMIM:618160'],
 'CLNDN': ['ISOLATED_GROWTH_HORMONE_DEFICIENCY', '_TYPE_V'],
 'CLNHGVS': ['NC_000001.10:g.104094352C>T'],
 'CLNREVSTAT': ['no_assertion_criteria_provided'],
 'CLNSIG': ['Pathogenic'],
 'CLNVC': 'single_nucleotide_variant',
 'CLNVCSO': 'SO:0001483',
 'CLNVI': ['OMIM_Allelic_Variant:618016.0002'],
 'GENEINFO': 'RNPC3:55599',
 'MC': ['SO:0001587|nonsense'],
 'ORIGIN': ['1']}

In [29]:
import gzip
with  gzip.open('/home/felipe/dbNSFP4.0b2a.txt.gz','rt') as f:
     for line in f:
            a=line
            break

In [37]:
a.split("\t")


['#chr',
 'pos(1-based)',
 'ref',
 'alt',
 'aaref',
 'aaalt',
 'rs_dbSNP151',
 'hg19_chr',
 'hg19_pos(1-based)',
 'hg18_chr',
 'hg18_pos(1-based)',
 'aapos',
 'genename',
 'Ensembl_geneid',
 'Ensembl_transcriptid',
 'Ensembl_proteinid',
 'Uniprot_acc',
 'Uniprot_entry',
 'APPRIS',
 'GENCODE_basic',
 'TSL',
 'VEP_canonical',
 'cds_strand',
 'refcodon',
 'codonpos',
 'codon_degeneracy',
 'Ancestral_allele',
 'AltaiNeandertal',
 'Denisova',
 'VindijiaNeandertal',
 'SIFT_score',
 'SIFT_converted_rankscore',
 'SIFT_pred',
 'SIFT4G_score',
 'SIFT4G_converted_rankscore',
 'SIFT4G_pred',
 'Polyphen2_HDIV_score',
 'Polyphen2_HDIV_rankscore',
 'Polyphen2_HDIV_pred',
 'Polyphen2_HVAR_score',
 'Polyphen2_HVAR_rankscore',
 'Polyphen2_HVAR_pred',
 'LRT_score',
 'LRT_converted_rankscore',
 'LRT_pred',
 'LRT_Omega',
 'MutationTaster_score',
 'MutationTaster_converted_rankscore',
 'MutationTaster_pred',
 'MutationTaster_model',
 'MutationTaster_AAE',
 'MutationAssessor_score',
 'MutationAssessor_ranksc

In [228]:

vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123 _cf_GRCh38.vcf.gz',compressed=True,encoding='UTF-8')


patcnt=0
for record in vcf_reader:
    if 'CLNSIG' in record.INFO.keys():
        if record.INFO['CLNSIG']==['Pathogenic']:
            taver=record
            break
            
 


In [230]:
taver.INFO


{'ALLELEID': 181485,
 'CLNDISDB': ['MedGen:C4015293', 'OMIM:616126', 'Orphanet:ORPHA319563'],
 'CLNDN': ['Immunodeficiency_38_with_basal_ganglia_calcification'],
 'CLNHGVS': ['NC_000001.11:g.1014143C>T'],
 'CLNREVSTAT': ['no_assertion_criteria_provided'],
 'CLNSIG': ['Pathogenic'],
 'CLNVC': 'single_nucleotide_variant',
 'CLNVCSO': 'SO:0001483',
 'CLNVI': ['OMIM_Allelic_Variant:147571.0003'],
 'GENEINFO': 'ISG15:9636',
 'MC': ['SO:0001587|nonsense'],
 'ORIGIN': ['1'],
 'RS': ['786201005']}

In [36]:
import re
vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123 _cf_GRCh38.vcf.gz',compressed=True,encoding='UTF-8')

clndb_id=[]
clndb= []
clnpath=[]
for record in vcf_reader:
    s="0"
    if 'CLNSIG' in record.INFO.keys():
        item=record.INFO['CLNSIG'][0]
        if item==None:
            continue
        s=item
    clnpath.append(s)
    clndb_id.append(int(record.ID))
    
    
    
 


df_cln_info_pat=pd.DataFrame({"clinvar_id":clndb_id,"patho_status":clnpath})

In [40]:
df_cln_info_pat.loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]

clinvar_id                                  patho_status
2           183381                                    Pathogenic
8           161455                                    Pathogenic
9           161454                                    Pathogenic
20          210112  Conflicting_interpretations_of_pathogenicity
24          243036                                    Pathogenic
26          243037                                    Pathogenic
31          210111  Conflicting_interpretations_of_pathogenicity
38          430121  Conflicting_interpretations_of_pathogenicity
46          574478                                    Pathogenic
49          126556                                    Pathogenic
63          243038                                    Pathogenic
135         235570  Conflicting_interpretations_of_pathogenicity
186         210108  Conflicting_interpretations_of_pathogenicity
237         489335                             Likely_pathogenic
239         252808  Conflicting_interpretations_of_pathogenicity
270         243039                                    Pathogenic
280          18241                                    Pathogenic
282         126555                                    Pathogenic
292         503670                             Likely_pathogenic
298         128314  Conflicting_interpretations_of_pathogenicity
314         243040                                    Pathogenic
365          96692                                    Pathogenic
371          60484                                    Pathogenic
375          60493                                    Pathogenic
387          60488                                    Pathogenic
388          60489                                    Pathogenic
390         452846                             Likely_pathogenic
391          60497                                    Pathogenic
393          60490                                    Pathogenic
398          60494                                    Pathogenic
...            ...                                           ...
440712      143885                             Likely_pathogenic
440713      143886                             Likely_pathogenic
440714      143887                             Likely_pathogenic
440715      143888                             Likely_pathogenic
440716      143924                             Likely_pathogenic
440719      143925                             Likely_pathogenic
440720      143889                             Likely_pathogenic
440721      143890                             Likely_pathogenic
440722      143891                             Likely_pathogenic
440727        9684                                    Pathogenic
440730      143894                             Likely_pathogenic
440731        9677                                    Pathogenic
440732        9683                                    Pathogenic
440733      143926                             Likely_pathogenic
440734        9678                                    Pathogenic
440735      143895                             Likely_pathogenic
440736      143896                             Likely_pathogenic
440737      143927                             Likely_pathogenic
440738      143897                             Likely_pathogenic
440741      143898                             Likely_pathogenic
440743      143899                             Likely_pathogenic
440744        9675                                    Pathogenic
440747      143900                             Likely_pathogenic
440749      143901                             Likely_pathogenic
440751      143902                             Likely_pathogenic
440752      143903                             Likely_pathogenic
440754      143904                             Likely_pathogenic
440755      143905                             Likely_pathogenic
440758        9572                                    Pathogenic
440759        9570                         

In [39]:
df_cln_info_pat["patho_status"].unique()

array(['Benign', 'Uncertain_significance', 'Pathogenic', 'Likely_benign',
       'Benign/Likely_benign',
       'Conflicting_interpretations_of_pathogenicity',
       'Likely_pathogenic', 'not_provided',
       'Pathogenic/Likely_pathogenic', 'risk_factor', '0', 'association',
       'drug_response', 'Affects', 'other', 'protective'], dtype=object)

In [27]:
sum(mer_omim_clin["auto_recessive"]|mer_omim_clin["auto_dominant"]|mer_omim_clin["x_recessive"]|mer_omim_clin["x_dominant"])

23866

Es decir hay 23866 variantes de un total de 85041 probablemente patogenicas (109841 posiblemente patogenicas) que tienen anotaciones de zigosidad

In [28]:
import re
vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123 _cf_GRCh38.vcf.gz',compressed=True,encoding='UTF-8')

clndb_id=[]
clndb= []
clnpath=[]
for record in vcf_reader:
    s=[]
    if 'CLNSIG' in record.INFO.keys():
         for item in record.INFO['CLNSIG']:
                if item==None:
                    continue
                s.append(item)
    clnpath.append(s)
    clndb_id.append(int(record.ID))
    
    
    
 


df_cln_info_pat_vec=pd.DataFrame({"clinvar_id":clndb_id,"patho_status":clnpath})

In [29]:
df_cln_info_pat_vec["num_anot"]=df_cln_info_pat_vec["patho_status"].apply(len)

In [35]:
df_cln_info_pat_vec.loc[df_cln_info_pat_vec["num_anot"]==3]

clinvar_id                                       patho_status  \
3062          3521  [Conflicting_interpretations_of_pathogenicity,...   
19246        18395            [Pathogenic, _association, _protective]   
79453        12274  [Conflicting_interpretations_of_pathogenicity,...   
140413          10                 [Pathogenic, _other, _risk_factor]   
175751        1552  [Conflicting_interpretations_of_pathogenicity,...   
201684        5182  [Benign/Likely_benign, _protective, _risk_factor]   
228758        8678  [Benign/Likely_benign, _drug_response, _risk_f...   
267587       18390         [drug_response, _protective, _risk_factor]   
330055        2211            [Likely_benign, _drug_response, _other]   
408145       13479  [Conflicting_interpretations_of_pathogenicity,...   
419001       16893            [Likely_benign, _drug_response, _other]   

        num_anot  
3062           3  
19246          3  
79453          3  
140413         3  
175751         3  
201684         3  
228758         3  
267587         3  
330055         3  
408145         3  
419001         3

In [41]:
len(mer_omim_clin)

440760

In [42]:
len(df_cln_info_pat)

440760

In [ ]:
len(df_med)

In [3]:
mer_omim_clin.loc[mer_omim_clin["clinvar_id"].isin(aver3)]

NameError: name 'aver3' is not defined

In [11]:
sum(mer_omim_clin_rare["x_dominant"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver3)]|mer_omim_clin_rare["x_recessive"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver3)]|mer_omim_clin_rare["auto_recessive"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver3)]|mer_omim_clin_rare["auto_dominant"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver3)])

363

In [12]:
len(aver3)

6881

In [16]:
sum(mer_omim_clin_rare["x_dominant"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver2)]|mer_omim_clin_rare["x_recessive"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver2)]|mer_omim_clin_rare["auto_recessive"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver2)]|mer_omim_clin_rare["auto_dominant"].loc[mer_omim_clin_rare["clinvar_id"].isin(aver2)])

354

In [82]:
pd_tuvieja=pd.concat([df_clnv_medgen_dominant,df_clnv_medgen_recessive,df_clnv_medgen_xdominant,df_clnv_medgen_xrecessive],sort=False)

In [83]:
sum((df_cln_info_pat['clinvar_id'].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]).isin(pd_tuvieja['clinvar_id'].unique()))

75925

In [84]:
df_cln_in_pos_pat=df_cln_info_pat.loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]

In [85]:
len(df_cln_in_pos_pat["clinvar_id"].loc[df_cln_in_pos_pat["clinvar_id"].isin(pd_tuvieja['clinvar_id'].values)])

75925

In [71]:
df_cln_info_pat['clinvar_id'].loc[]

IndexingError: Unalignable boolean Series provided as indexer (index of the boolean Series and of the indexed object do not match

In [86]:

df_clnv_medgen_dominant=pd.read_csv("clinvar_medgen_dominant_result.txt",sep="\t")

df_clnv_medgen_dominant.rename(columns={"VariationID":"clinvar_id"},inplace=True)


df_cln_info_pat["auto_dominant"]=df_cln_info_pat["clinvar_id"].isin(df_clnv_medgen_dominant["clinvar_id"].values)

df_clnv_medgen_recessive=pd.read_csv("clinvar_medgen_recessive_result.txt",sep="\t")

df_clnv_medgen_recessive.rename(columns={"VariationID":"clinvar_id"},inplace=True)

df_cln_info_pat["auto_recessive"]=df_cln_info_pat["clinvar_id"].isin(df_clnv_medgen_recessive["clinvar_id"].values)




df_clnv_medgen_xrecessive=pd.read_csv("clinvar_medgen_xrecessive_result.txt",sep="\t")

df_clnv_medgen_xrecessive.rename(columns={"VariationID":"clinvar_id"},inplace=True)



df_cln_info_pat["x_recessive"]=df_cln_info_pat["clinvar_id"].isin(df_clnv_medgen_xrecessive["clinvar_id"].values)





df_clnv_medgen_xdominant=pd.read_csv("clinvar_medgen_xdominant_result.txt",sep="\t")

df_clnv_medgen_xdominant.rename(columns={"VariationID":"clinvar_id"},inplace=True)

df_cln_info_pat["x_dominant"]=df_cln_info_pat["clinvar_id"].isin(df_clnv_medgen_xdominant["clinvar_id"].values)


In [87]:
sum(df_cln_info_pat["auto_recessive"]|df_cln_info_pat["auto_dominant"]|df_cln_info_pat["x_recessive"]|df_cln_info_pat["x_dominant"])

239310

In [88]:
sum((df_cln_info_pat["auto_recessive"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])])|(df_cln_info_pat["auto_dominant"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])])|(df_cln_info_pat["x_recessive"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])])|(df_cln_info_pat["x_dominant"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]))

75925

que significa que tenga una clasificacion de herencia pero no una de patogenica

In [89]:
df_cln_info_pat["has_any_classification"]=df_cln_info_pat["auto_recessive"]|df_cln_info_pat["auto_dominant"]|df_cln_info_pat["x_recessive"]|df_cln_info_pat["x_dominant"]

In [108]:
sum((df_cln_info_pat["has_any_classification"].loc[~df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]))

163385

estos 163385 tienen marca de herencias pero no estan en putativamente patogenicos.Chequee a mano un par y parece bien...tienen baja asocaicion a una enfermedad pero si se conoce la forma de herencia de la enfermedad

In [110]:
df_cln_info_pat.loc[~df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]

clinvar_id            patho_status  auto_dominant  auto_recessive  \
0           475283                  Benign          False            True   
1           542074  Uncertain_significance          False            True   
3           542075  Uncertain_significance          False            True   
4           475278                  Benign          False            True   
5           402986                  Benign          False           False   
6           571208  Uncertain_significance          False            True   
7           568195  Uncertain_significance          False            True   
10          542076           Likely_benign          False            True   
11          475281                  Benign          False            True   
12          475282           Likely_benign          False            True   
13          474165  Uncertain_significance          False            True   
14          387476    Benign/Likely_benign          False            True   
15          574825  Uncertain_significance          False            True   
16          377270  Uncertain_significance          False           False   
17          541157  Uncertain_significance          False            True   
18          128310                  Benign          False           False   
19          388958  Uncertain_significance          False           False   
21          571135  Uncertain_significance          False            True   
22          581563  Uncertain_significance          False            True   
23          263166                  Benign          False           False   
25          128296                  Benign          False            True   
27          541155  Uncertain_significance          False            True   
28          541203           Likely_benign          False            True   
29          474141           Likely_benign          False            True   
30          541188           Likely_benign          False            True   
32          474159                  Benign          False            True   
33          474163  Uncertain_significance          False            True   
34          541197           Likely_benign          False            True   
35          541156  Uncertain_significance          False            True   
36          263202                  Benign          False            True   
...            ...                     ...            ...             ...   
440652      441110            not_provided          False           False   
440656      370060  Uncertain_significance          False           False   
440662      374216  Uncertain_significance          False            True   
440670      370061  Uncertain_significance          False           False   
440673      235655           Likely_benign          False           False   
440685      445709           Likely_benign          False           False   
440686      370063  Uncertain_significance          False           False   
440688      370064  Uncertain_significance          False           False   
440691      445359           Likely_benign          False           False   
440694      235375           Likely_benign          False           False   
440696      377041                  Benign          False           False   
440701      376959           Likely_benign          False           False   
440717      235525  Uncertain_significance          False           False   
440718      373995  Uncertain_significance          False           False   
440723      143892                  Benign           True           False   
440724      377154           Likely_benign          False           False   
440725      441132            not_provided          False           False   
440726        9687           Likely_benign           True            True   
440728      143893           Likely_benign           True           False   
440729      235650           Likely_benign          False           False   
4407

In [96]:
import re
vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123 _cf_GRCh38.vcf.gz',compressed=True,encoding='UTF-8')

clndb_id=[]
clndb= []
clnpath=[]
for record in vcf_reader:
    s="0"
    if 'CLNDISDB' in record.INFO.keys():
         for item in record.INFO['CLNDISDB']:
                if item==None:
                    continue
                if len(re.findall(r"OMIM:",item)):
                    s=re.search(r"OMIM:(\d+)",item).group(1)
    clndb.append(int(s))
    clndb_id.append(int(record.ID))
    
    
    
 


df_cln_db=pd.DataFrame({"clinvar_id":clndb_id,"Mim Number":clndb})

In [98]:
df_omim=pd.read_csv("OMIM_genemap2.txt",sep="\t",engine='python',skiprows=3)

df_omim=df_omim.loc[df_omim["Mim Number"].notna()]

df_omim["Mim Number"]=df_omim["Mim Number"].astype(pd.Int64Dtype(),errors="ignore",inplace=True)

df_omim["Comments"]=df_omim["Comments"].str.upper()
df_omim["Phenotypes"]=df_omim["Phenotypes"].str.upper()

df_omim["auto_recessive"]=((df_omim["Phenotypes"].str.extract(r'(AUTOSOMAL RECESSIVE)')=="AUTOSOMAL RECESSIVE")|(df_omim["Comments"].str.extract(r'(AUTOSOMAL RECESSIVE)')=="AUTOSOMAL RECESSIVE")) 


df_omim["auto_dominant"]=((df_omim["Phenotypes"].str.extract(r'(AUTOSOMAL DOMINANT)')=="AUTOSOMAL DOMINANT")|(df_omim["Comments"].str.extract(r'(AUTOSOMAL DOMINANT)')=="AUTOSOMAL DOMINANT")) 


mer_omim_clin=df_omim.merge(df_cln_db,how="right",on=["Mim Number"])



In [102]:

df_cln_info_pat["auto_dominant"]=df_cln_info_pat["auto_dominant"]|df_cln_info_pat["clinvar_id"].isin(mer_omim_clin["clinvar_id"].loc[mer_omim_clin["auto_dominant"]==True])


In [103]:

df_cln_info_pat["auto_recessive"]=df_cln_info_pat["auto_recessive"]|df_cln_info_pat["clinvar_id"].isin(mer_omim_clin["clinvar_id"].loc[mer_omim_clin["auto_recessive"]==True])


In [112]:
df_cln_info_pat["has_any_classification"]=df_cln_info_pat["auto_recessive"]|df_cln_info_pat["auto_dominant"]|df_cln_info_pat["x_recessive"]|df_cln_info_pat["x_dominant"]

In [113]:
sum((df_cln_info_pat["auto_recessive"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])])|(df_cln_info_pat["auto_dominant"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])])|(df_cln_info_pat["x_recessive"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])])|(df_cln_info_pat["x_dominant"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]))

75947

In [114]:
sum((df_cln_info_pat["has_any_classification"].loc[df_cln_info_pat["patho_status"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]))

75947

In [115]:
sum(df_cln_info_pat["auto_recessive"]|df_cln_info_pat["auto_dominant"]|df_cln_info_pat["x_recessive"]|df_cln_info_pat["x_dominant"])

239350

In [107]:
sum(df_cln_info_pat["auto_recessive"])

131475

aportan 23 nuevos nada mas de los probablementes patogenicos y 50 de los otros

In [106]:
df_cln_info_pat.to_csv("df_cln_info_pat")

In [101]:
mer_omim_clin

# Chromosome  Genomic Position Start  Genomic Position End  \
0              chr1                     0.0            27600000.0   
1              chr1                     0.0            27600000.0   
2              chr1                     0.0            27600000.0   
3              chr1                     0.0            27600000.0   
4              chr1                     0.0            27600000.0   
5              chr1              60800000.0           106700000.0   
6              chr1              60800000.0            84400000.0   
7              chr1              84400000.0            94300000.0   
8              chr1             165500000.0           173000000.0   
9              chr1             185800000.0           236400000.0   
10             chr1             185800000.0           236400000.0   
11             chr1             185800000.0           236400000.0   
12             chr1             185800000.0           236400000.0   
13             chr1             185800000.0           236400000.0   
14             chr1             185800000.0           236400000.0   
15             chr1             185800000.0           236400000.0   
16             chr1             185800000.0           236400000.0   
17             chr1             185800000.0           236400000.0   
18             chr1             185800000.0           236400000.0   
19             chr1             185800000.0           236400000.0   
20             chr1             185800000.0           236400000.0   
21             chr1             185800000.0           236400000.0   
22             chr1             185800000.0           236400000.0   
23             chr1             185800000.0           236400000.0   
24             chr1             185800000.0           236400000.0   
25             chr1             185800000.0           236400000.0   
26             chr1             185800000.0           236400000.0   
27             chr1             185800000.0           236400000.0   
28             chr1             185800000.0           236400000.0   
29             chr1             185800000.0           236400000.0   
...             ...                     ...                   ...   
440730          NaN                     NaN                   NaN   
440731          NaN                     NaN                   NaN   
440732          NaN                     NaN                   NaN   
440733          NaN                     NaN                   NaN   
440734          NaN                     NaN                   NaN   
440735          NaN                     NaN                   NaN   
440736          NaN                     NaN                   NaN   
440737          NaN                     NaN                   NaN   
440738          NaN                     NaN                   NaN   
440739          NaN                     NaN                   NaN   
440740          NaN                     NaN                   NaN   
440741          NaN                     NaN                   NaN   
440742          NaN                     NaN                   NaN   
440743          NaN                     NaN                   NaN   
440744          NaN                     NaN                   NaN   
440745          NaN                     NaN                   NaN   
440746          NaN                     NaN                   NaN   
440747          NaN                     NaN                   NaN   
440748          NaN                     NaN                   NaN   
440749          NaN                     NaN                   NaN   
440750          NaN                     NaN                   NaN   
440751          NaN                     NaN                   NaN   
440752          NaN                     NaN                   NaN   
440753          NaN                     NaN                   NaN   
440754          NaN                     NaN                   NaN   
440755          NaN                     NaN                   NaN   
44075

In [17]:
df_rare_pat=rare_variants_dbfnsp[["clinvar_id","clinvar_clnsig","Ensembl_geneid"]].loc[(rare_variants_dbfnsp["clinvar_clnsig"].isin(["Pathogenic","Likely_pathogenic"]))]



In [239]:
df_rare_pat["clinvar_id"]=df_rare_pat["clinvar_id"].astype(pd.Int64Dtype(),errors="ignore",inplace=True)

In [240]:
df_rare_pat=df_rare_pat.merge(mer_omim_clin_rare,how="left",on=["clinvar_id"])

In [241]:

df_rare_pat["x_recessive"]=df_rare_pat["clinvar_id"].isin(df_clnv_medgen_xrecessive["clinvar_id"].values)


In [242]:

df_rare_pat["x_dominant"]=df_rare_pat["clinvar_id"].isin(df_clnv_medgen_xdominant["clinvar_id"].values)


In [243]:

df_rare_pat["auto_dominant"]=df_rare_pat["clinvar_id"].isin(df_clnv_medgen_dominant["clinvar_id"].values)|df_rare_pat["auto_dominant"]


In [244]:

df_rare_pat["auto_recessive"]=df_rare_pat["clinvar_id"].isin(df_clnv_medgen_recessive["clinvar_id"].values)|df_rare_pat["auto_recessive"]


In [245]:
sum(df_rare_pat["auto_recessive"]|df_rare_pat["auto_dominant"]|df_rare_pat["x_recessive"]|df_rare_pat["x_dominant"])

8297

In [246]:
sum(df_rare_pat["auto_recessive"]|df_rare_pat["x_recessive"]|df_rare_pat["x_dominant"])

7108

In [247]:
sum(df_rare_pat["auto_recessive"])

6988

In [18]:
len(df_rare_pat)

11042

de 11042 patogenicas o posiblemente patogenicas raras solo tenemos data de homozigosidad para 8297.Chequeando de a una se puede ver que en general corresponden a variantes con muy pocos reportes. 

In [ ]:
df_rare_pat.to_csv("df_rare_pat")

In [119]:
rare_variants_dbfnsp=pd.read_csv("rare_control_variants_hom_dbNSFP4.0b2a.txt.tsv",sep="\t")

/home/felipe/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,7,9,11,18,20,25,26,30,33,35,36,38,39,41,44,46,48,49,50,51,53,54,56,57,60,74,76,77,78,79,81,85,86,88,89,90,91,92,93,94,106,219,292,358,359,360,361,362,363,364,365,366) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [126]:
import re
vcf_reader = vcf.Reader( filename='/home/felipe/clinvar/clinvar_20190123 _cf_GRCh38.vcf.gz',compressed=True,encoding='UTF-8')

clndb_id=[]
clndb= []
clnpath=[]
for record in vcf_reader:
    s=[]
    if 'CLNREVSTAT' in record.INFO.keys():
         for item in record.INFO['CLNREVSTAT']:
                if item==None:
                    continue
                s.append(item)
    clnpath.append(s)
    clndb_id.append(int(record.ID))
    
    
    
 


df_cln_info_rev_vec=pd.DataFrame({"clinvar_id":clndb_id,"patho_rev":clnpath})

df_cln_info_rev_vec["num_rev"]=df_cln_info_rev_vec["patho_rev"].apply(len)

df_cln_info_rev_vec.loc[df_cln_info_rev_vec["num_rev"]==3]

clinvar_id                                          patho_rev  num_rev
14          387476  [criteria_provided, _multiple_submitters, _no_...        3
18          128310  [criteria_provided, _multiple_submitters, _no_...        3
23          263166  [criteria_provided, _multiple_submitters, _no_...        3
25          128296  [criteria_provided, _multiple_submitters, _no_...        3
36          263202  [criteria_provided, _multiple_submitters, _no_...        3
37          263203  [criteria_provided, _multiple_submitters, _no_...        3
39          263204  [criteria_provided, _multiple_submitters, _no_...        3
40          128320  [criteria_provided, _multiple_submitters, _no_...        3
50          128291  [criteria_provided, _multiple_submitters, _no_...        3
52          263158  [criteria_provided, _multiple_submitters, _no_...        3
57          128292  [criteria_provided, _multiple_submitters, _no_...        3
65          263160  [criteria_provided, _multiple_submitters, _no_...        3
66          263161  [criteria_provided, _multiple_submitters, _no_...        3
71          263162  [criteria_provided, _multiple_submitters, _no_...        3
74          263163  [criteria_provided, _multiple_submitters, _no_...        3
80          263164  [criteria_provided, _multiple_submitters, _no_...        3
83          263165  [criteria_provided, _multiple_submitters, _no_...        3
94          210106  [criteria_provided, _multiple_submitters, _no_...        3
97          128293  [criteria_provided, _multiple_submitters, _no_...        3
100         263167  [criteria_provided, _multiple_submitters, _no_...        3
105         128294  [criteria_provided, _multiple_submitters, _no_...        3
106         541148  [criteria_provided, _multiple_submitters, _no_...        3
107         263168  [criteria_provided, _multiple_submitters, _no_...        3
108         263169  [criteria_provided, _multiple_submitters, _no_...        3
109         263170  [criteria_provided, _multiple_submitters, _no_...        3
115         128295  [criteria_provided, _multiple_submitters, _no_...        3
117         190974  [criteria_provided, _multiple_submitters, _no_...        3
120         387869  [criteria_provided, _multiple_submitters, _no_...        3
126         263173  [criteria_provided, _multiple_submitters, _no_...        3
134         263176  [criteria_provided, _multiple_submitters, _no_...        3
...            ...                                                ...      ...
439682       37203  [criteria_provided, _multiple_submitters, _no_...        3
439684       10393  [criteria_provided, _multiple_submitters, _no_...        3
439685       10367  [criteria_provided, _multiple_submitters, _no_...        3
439705       10402  [criteria_provided, _multiple_submitters, _no_...        3
439708      521562  [criteria_provided, _multiple_submitters, _no_...        3
439769      459584  [criteria_provided, _multiple_submitters, _no_...        3
439795      166994  [criteria_provided, _multiple_submitters, _no_...        3
439801      377149  [criteria_provided, _multiple_submitters, _no_...        3
439835      128896  [criteria_provided, _multiple_submitters, _no_...        3
439836      210847  [criteria_provided, _multiple_submitters, _no_...        3
439843      166995  [criteria_provided, _multiple_submitters, _no_...        3
439853      368114  [criteria_provided, _multiple_submitters, _no_...        3
439868       41001  [criteria_provided, _multiple_submitters, _no_...        3
439954      255223  [criteria_provided, _multiple_submitters, _no_...        3
439962       41000  [criteria_provided, _multiple_submitters, _no_...        3
439963       40999  [criteria_provided, _multiple_submitters, _no_...        3
439990       10236  [criteria_provided, _multiple_submitters, _no_...        3
440003       10226  [criteria_provided, _multiple_submitters, _no_...        3
440033      368124  [criteria_provided, _multiple_submitte

In [136]:
df_cln_info_pat=df_cln_info_pat.merge(df_cln_info_rev_vec,on=["clinvar_id"])

In [137]:
df_cln_info_pat.to_csv("df_cln_info_pat")

In [142]:
set(set([a for b in df_cln_info_pat["patho_rev"].tolist() for a in b]))

{'_conflicting_interpretations',
 '_multiple_submitters',
 '_no_conflicts',
 '_single_submitter',
 'criteria_provided',
 'no_assertion_criteria_provided',
 'no_assertion_provided',
 'no_interpretation_for_the_single_variant',
 'practice_guideline',
 'reviewed_by_expert_panel'}

In [248]:
import igraph



In [249]:
biogrid=pd.read_csv("BIOGRID/BIOGRID-ORGANISM-3.5.168.tab2/BIOGRID-ORGANISM-Homo_sapiens-3.5.168.tab2.txt",low_memory=False,sep="\t")

In [217]:
biogrid

#BioGRID Interaction ID  Entrez Gene Interactor A  \
0                           103                      6416   
1                           117                     84665   
2                           183                        90   
3                           278                      2624   
4                           418                      6118   
5                           586                       375   
6                           612                       377   
7                           617                       377   
8                           663                     54464   
9                           866                       351   
10                          917                       333   
11                         1156                     10370   
12                         1161                      2033   
13                         1206                       338   
14                         1427                       409   
15                         1611                      1436   
16                         1695                      7916   
17                         1783                     27257   
18                         1888                      6521   
19                         1958                       602   
20                         2006                       153   
21                         2368                       672   
22                         2398                       672   
23                         2411                       672   
24                         2424                       672   
25                         2466                       421   
26                         2721                      5092   
27                         2765                      5664   
28                         2785                       825   
29                         2827                      3708   
...                         ...                       ...   
463074                  2521051                     55023   
463075                  2521052                     55023   
463076                  2521053                     81620   
463077                  2521054                     81620   
463078                  2521055                     81620   
463079                  2521067                     56648   
463080                  2521068                      1028   
463081                  2521069                      1028   
463082                  2521070                      1028   
463083                  2521071                      3091   
463084                  2521072                     11168   
463085                  2521073                     10524   
463086                  2521074                     11168   
463087                  2521075                       405   
463088                  2521076                      8350   
463089                  2521077                      1616   
463090                  2521078                      1616   
463091                  2521079                      3020   
463092                  2521080                      3091   
463093                  2521081                      5599   
463094                  2521082                      2033   
463095                  2521083                      3091   
463096                  2521085                       405   
463097                  2521086                       405   
463098                  2521087                       405   
463099                  2521088                       405   
463100                  2521089                       405   
463101                  2521090                       405   
463102                  2521091                       405   
463103                  2521092                       405   

        Entrez Gene Interactor B  BioGRID ID Interactor A  \
0                           2318                   112315   
1                             88                   124185   
2                           2339          

In [218]:
biogrid.columns

Index(['#BioGRID Interaction ID', 'Entrez Gene Interactor A',
       'Entrez Gene Interactor B', 'BioGRID ID Interactor A',
       'BioGRID ID Interactor B', 'Systematic Name Interactor A',
       'Systematic Name Interactor B', 'Official Symbol Interactor A',
       'Official Symbol Interactor B', 'Synonyms Interactor A',
       'Synonyms Interactor B', 'Experimental System',
       'Experimental System Type', 'Author', 'Pubmed ID',
       'Organism Interactor A', 'Organism Interactor B', 'Throughput', 'Score',
       'Modification', 'Phenotypes', 'Qualifications', 'Tags',
       'Source Database'],
      dtype='object')

In [ ]:
Entrez_gene_id

In [220]:
df_biogrid=biogrid[['Entrez Gene Interactor A','Entrez Gene Interactor B']]

In [221]:
tuples = [tuple(x) for x in df_biogrid.values]
Gm = igraph.Graph.TupleList(tuples, directed = True, edge_attrs = ['weight'])

In [231]:
rare_variants_dbfnsp.columns[:30]

Index(['#chr', 'pos(1-based)', 'ref', 'alt', 'aaref', 'aaalt', 'rs_dbSNP151',
       'hg19_chr', 'hg19_pos(1-based)', 'hg18_chr', 'hg18_pos(1-based)',
       'aapos', 'genename', 'Ensembl_geneid', 'Ensembl_transcriptid',
       'Ensembl_proteinid', 'Uniprot_acc', 'Uniprot_entry', 'APPRIS',
       'GENCODE_basic', 'TSL', 'VEP_canonical', 'cds_strand', 'refcodon',
       'codonpos', 'codon_degeneracy', 'Ancestral_allele', 'AltaiNeandertal',
       'Denisova', 'VindijiaNeandertal'],
      dtype='object')

In [280]:
dbfsnp_gen= pd.read_csv('dbNSFP4.0b2_gene.gz', compression='gzip',sep="\t",low_memory=False)

In [281]:
dbfsnp_gen.columns

Index(['Gene_name', 'Ensembl_gene', 'chr', 'Gene_old_names',
       'Gene_other_names', 'Uniprot_acc(HGNC/Uniprot)',
       'Uniprot_id(HGNC/Uniprot)', 'Entrez_gene_id', 'CCDS_id', 'Refseq_id',
       'ucsc_id', 'MIM_id', 'Gene_full_name', 'Pathway(Uniprot)',
       'Pathway(BioCarta)_short', 'Pathway(BioCarta)_full',
       'Pathway(ConsensusPathDB)', 'Pathway(KEGG)_id', 'Pathway(KEGG)_full',
       'Function_description', 'Disease_description', 'MIM_phenotype_id',
       'MIM_disease', 'Trait_association(GWAS)', 'GO_biological_process',
       'GO_cellular_component', 'GO_molecular_function',
       'Tissue_specificity(Uniprot)', 'Expression(egenetics)',
       'Expression(GNF/Atlas)', 'Interactions(IntAct)',
       'Interactions(BioGRID)', 'Interactions(ConsensusPathDB)', 'P(HI)',
       'HIPred_score', 'HIPred', 'GHIS', 'P(rec)', 'Known_rec_info',
       'RVIS_EVS', 'RVIS_percentile_EVS', 'LoF-FDR_ExAC', 'RVIS_ExAC',
       'RVIS_percentile_ExAC', 'ExAC_pLI', 'ExAC_pRec', 'ExAC_pNu

In [282]:
dbfsnp_gen.rename(columns={"Ensembl_gene":"Ensembl_geneid"},inplace=True)

In [283]:
dbfsnp_gen

Gene_name   Ensembl_geneid chr             Gene_old_names  \
0             A1BG  ENSG00000121410  19                          .   
1         A1BG-AS1  ENSG00000268895  19  NCRNA00181;A1BGAS;A1BG-AS   
2             A1CF  ENSG00000148584  10                          .   
3              A2M  ENSG00000175899  12                          .   
4          A2M-AS1  ENSG00000245105  12                          .   
5            A2ML1  ENSG00000166535  12                     CPAMD9   
6        A2ML1-AS1  ENSG00000256661  12                          .   
7        A2ML1-AS2  ENSG00000256904  12                          .   
8            A2MP1  ENSG00000256069  12                       A2MP   
9          A3GALT2  ENSG00000184389   1                   A3GALT2P   
10          A4GALT  ENSG00000128274  22                         P1   
11           A4GNT  ENSG00000118017   3                          .   
12            AAAS  ENSG00000094914  12                          .   
13            AACS  ENSG00000081760  12                          .   
14          AACSP1  ENSG00000250420   5                      AACSL   
15           AADAC  ENSG00000114771   3                          .   
16         AADACL2  ENSG00000197953   3                          .   
17     AADACL2-AS1  ENSG00000242908   3                          .   
18         AADACL3  ENSG00000188984   1                          .   
19         AADACL4  ENSG00000204518   1                          .   
20         AADACP1  ENSG00000240602   3                          .   
21           AADAT  ENSG00000109576   4                          .   
22           AAGAB  ENSG00000103591  15                          .   
23            AAK1  ENSG00000115977   2                          .   
24           AAMDC  ENSG00000087884  11                   C11orf67   
25            AAMP  ENSG00000127837   2                          .   
26           AANAT  ENSG00000129673  17                          .   
27            AAR2  ENSG00000131043  20                    C20orf4   
28            AARD  ENSG00000205002   8                    C8orf85   
29            AARS  ENSG00000090861  16                          .   
...            ...              ...  ..                        ...   
41520      ZSCAN30  ENSG00000186814  18                   ZNF397OS   
41521      ZSCAN31  ENSG00000235109   6             ZNF310P;ZNF323   
41522      ZSCAN32  ENSG00000140987  16                     ZNF434   
41523       ZSWIM1  ENSG00000168612  20                  C20orf162   
41524       ZSWIM2  ENSG00000163012   2                          .   
41525       ZSWIM3  ENSG00000132801  20                  C20orf164   
41526       ZSWIM4  ENSG00000132003  19                          .   
41527       ZSWIM5  ENSG00000162415   1                          .   
41528     ZSWIM5P1  ENSG00000235430  17                          .   
41529     ZSWIM5P2  ENSG00000231258  17                          .   
41530     ZSWIM5P3  ENSG00000248958   4                          .   
41531       ZSWIM6  ENSG00000130449   5                          .   
41532       ZSWIM7  ENSG00000214941  17                          .   
41533       ZSWIM8  ENSG00000214655  10                   KIAA0913   
41534   ZSWIM8-AS1  ENSG00000272589  10                          .   
41535       ZSWIM9  ENSG00000185453  19                   C19orf68   
41536         ZUP1  ENSG00000153975   6             C6orf113;ZUFSP   
41537         ZW10  ENSG00000086827  11                          .   
41538       ZWILCH  ENSG00000174442  15                          .   
41539        ZWINT  ENSG00000122952  10                          .   
41540         ZXDA  ENSG00000198205   X                          .   
41541         ZXDB  ENSG00000198455   X                          .   
41542         ZXDC  ENSG00000070476   3                          .   
41543       ZYG11A  ENSG00000203995   1                          .   
41544     ZYG11AP1  ENSG00000232242   9                          .   
41545       ZYG11B 

In [284]:
df_ensem_to_entres=dbfsnp_gen[["Ensembl_geneid","Entrez_gene_id"]]

In [286]:
df_rare_pat=df_rare_pat.merge(df_ensem_to_entres,how="left",on=["Ensembl_geneid"])

In [266]:
df_rare_pat.columns

Index(['clinvar_id', 'clinvar_clnsig', 'Ensembl_geneid', '# Chromosome',
       'Genomic Position Start', 'Genomic Position End', 'Cyto Location',
       'Computed Cyto Location', 'Mim Number', 'Gene Symbols', 'Gene Name',
       'Approved Symbol', 'Entrez Gene ID', 'Ensembl Gene ID', 'Comments',
       'Phenotypes', 'Mouse Gene Symbol/ID', 'Phenotypes]', 'auto_recessive',
       'auto_dominant', 'x_recessive', 'x_dominant'],
      dtype='object')

In [294]:
seq=Gm.vs["name"]

In [306]:
seq.

[6416,
 2318,
 84665,
 88,
 90,
 2339,
 2624,
 5371,
 6118,
 6774,
 375,
 23163,
 377,
 23647,
 27236,
 54464,
 226,
 351,
 10513,
 333,
 1600,
 10370,
 7020,
 2033,
 338,
 4547,
 409,
 5900,
 1436,
 2885,
 7916,
 27257,
 4677,
 6521,
 22950,
 602,
 580,
 153,
 10755,
 672,
 466,
 4436,
 2956,
 421,
 1013,
 5092,
 775,
 5664,
 823,
 825,
 7273,
 3708,
 767,
 9223,
 1499,
 5925,
 1523,
 7251,
 1026,
 4998,
 4171,
 5000,
 4174,
 8317,
 4999,
 10926,
 10270,
 6117,
 6119,
 5001,
 4172,
 7534,
 993,
 81704,
 998,
 9322,
 1033,
 1017,
 5932,
 5933,
 9464,
 1059,
 1213,
 9146,
 3486,
 1277,
 7295,
 1386,
 4088,
 4089,
 896,
 27335,
 10928,
 891,
 5479,
 819,
 1465,
 1445,
 3643,
 3480,
 1634,
 1956,
 6493,
 405,
 6492,
 266743,
 1839,
 7704,
 5111,
 1647,
 11073,
 7155,
 2316,
 1814,
 27258,
 6635,
 25804,
 23658,
 1958,
 1387,
 4926,
 2035,
 54512,
 1915,
 5335,
 3895,
 1936,
 3791,
 308,
 133522,
 2099,
 6597,
 79039,
 30836,
 2260,
 2224,
 7040,
 7341,
 355,
 51678,
 2512,
 2243,
 3688,
 

In [296]:
Gm.vs["pathogenicity"]=[id in df_rare_pat["Entrez_gene_id"] for id in Gm.vs["name"]   ]

In [299]:
Gm.vs["pathogenicity"]

[True,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 False,
 False,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 True,
 False,
 True,
 False,
 False,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 True,
 True,
 True,
 True,
 False,
 True,
 True,
 False,
 False,
 True,
 True,
 True,
 True,
 True,
 False,
 

In [307]:
v_patho=[vertex["name"] for vertex in Gm.vs if vertex["pathogenicity"]]
v_patho

[6416,
 2318,
 88,
 90,
 2339,
 2624,
 5371,
 6118,
 6774,
 375,
 377,
 226,
 351,
 10513,
 333,
 1600,
 10370,
 7020,
 2033,
 338,
 4547,
 409,
 5900,
 1436,
 2885,
 7916,
 4677,
 6521,
 602,
 580,
 153,
 10755,
 672,
 466,
 4436,
 2956,
 421,
 1013,
 5092,
 775,
 5664,
 823,
 825,
 7273,
 3708,
 767,
 9223,
 1499,
 5925,
 1523,
 7251,
 1026,
 4998,
 4171,
 5000,
 4174,
 8317,
 4999,
 10926,
 10270,
 6117,
 6119,
 5001,
 4172,
 7534,
 993,
 998,
 9322,
 1033,
 1017,
 5932,
 5933,
 9464,
 1059,
 1213,
 9146,
 3486,
 1277,
 7295,
 1386,
 4088,
 4089,
 896,
 10928,
 891,
 5479,
 819,
 1465,
 1445,
 3643,
 3480,
 1634,
 1956,
 6493,
 405,
 6492,
 1839,
 7704,
 5111,
 1647,
 7155,
 2316,
 1814,
 6635,
 1958,
 1387,
 4926,
 2035,
 1915,
 5335,
 3895,
 1936,
 3791,
 308,
 2099,
 6597,
 2260,
 2224,
 7040,
 7341,
 355,
 2512,
 2243,
 3688,
 10817,
 2566,
 2810,
 2908,
 7041,
 2645,
 5208,
 3309,
 1801,
 6514,
 3838,
 6517,
 1616,
 2898,
 6386,
 9463,
 2897,
 904,
 2896,
 10287,
 2781,
 2775,


In [310]:
path_to_patho=[len(path)for path in Gm.vs[0].get_shortest_paths(v_patho)]

/home/felipe/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: RuntimeWarning: Couldn't reach some vertices at structural_properties.c:740
  """Entry point for launching an IPython kernel.


In [315]:
len_to_patho=list(set(path_to_patho))

In [317]:
len_to_patho.sort()

In [318]:
len_to_patho[1]

2

In [323]:
def get_min_dist_to_patho(Vtx):
    path_to_patho=[len(path)for path in Vtx.get_shortest_paths(v_patho)]
    len_to_patho=list(set(path_to_patho))

    len_to_patho.sort()
    if(len(len_to_patho)==1):
        return(-1000)

    return(len_to_patho[1])
    
    

In [324]:
Gm.vs["dist_to_patho"]=[ get_min_dist_to_patho(Vtx)  for Vtx in Gm.vs]

/home/felipe/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: RuntimeWarning: Couldn't reach some vertices at structural_properties.c:740
  


In [325]:
Gm.vs["dist_to_patho"]


[2,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 1,
 1,
 2,
 1,
 2,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 2,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 1,
 2,
 2,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 1,
 -1000,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 1,
 1,
 2,
 1,
 2,
 2,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 1,
 1,
 1,


In [330]:
Gm.neighbors(Gm.vs[0])

[0,
 0,
 1,
 1,
 1,
 7,
 17,
 26,
 26,
 78,
 101,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 207,
 277,
 287,
 287,
 287,
 287,
 287,
 344,
 566,
 566,
 628,
 628,
 628,
 818,
 818,
 818,
 850,
 855,
 855,
 855,
 1043,
 1043,
 1043,
 1703,
 1703,
 1703,
 1759,
 1970,
 1970,
 1970,
 1988,
 2083,
 2806,
 2847,
 2847,
 2847,
 3475,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3670,
 3907,
 3907,
 3907,
 3968,
 3968,
 3968,
 3968,
 4387,
 4387,
 4430,
 4575,
 4748,
 4769,
 6089,
 6089,
 6096,
 6096,
 6191,
 6224,
 6614,
 6748,
 6748,
 6820,
 7678,
 9374,
 17245]

In [334]:
def get_number_patho_first_neigh(Vtx):
    if(len(Gm.neighbors(Vtx))):
       return(0)
    sum(Gm.vs[list(set(Gm.neighbors(Vtx)))]["pathogenicity"])

In [335]:

Gm.vs["patho_first_neigh"]=[ get_number_patho_first_neigh(Vtx)  for Vtx in Gm.vs]





In [337]:
Gm.save("bio_grid_graph",format="picklez")

In [338]:
df_rare_pat.to_csv("dbfsnp_rare_patho-likely_patho")

In [380]:

vcf_reader = vcf.Reader( filename='/home/felipe/gnomad.exomes.r2.1.sites.vcf.bgz',compressed=True,encoding='UTF-8')


patcnt=0
for i,record in enumerate(vcf_reader):
    a=record
    if i>20000:
        if a.INFO['nhomalt'][0]!=0:
            break

 

In [381]:
a.INFO


{'AC': [89404],
 'AN': [243526],
 'AF': [0.367123],
 'rf_tp_probability': 0.940327,
 'FS': 0.0,
 'InbreedingCoeff': 0.1767,
 'MQ': 59.5,
 'MQRankSum': 0.099,
 'QD': 18.53,
 'ReadPosRankSum': 0.298,
 'SOR': 0.705,
 'VQSR_POSITIVE_TRAIN_SITE': True,
 'BaseQRankSum': -1.585,
 'ClippingRankSum': 0.0,
 'DP': 6432380,
 'VQSLOD': 5.76,
 'VQSR_culprit': 'FS',
 'rf_positive_label': True,
 'rf_label': 'TP',
 'rf_train': True,
 'variant_type': 'snv',
 'allele_type': ['snv'],
 'n_alt_alleles': [1],
 'pab_max': [1.0],
 'gq_hist_alt_bin_freq': ['56|223|185|310|396|310|501|618|493|909|1232|906|1466|1687|1130|1857|1781|1221|1750|71819'],
 'gq_hist_all_bin_freq': ['1283|1244|806|1518|1868|1253|2394|2797|1757|3432|4158|2607|6915|2966|5201|3557|6671|2142|6499|104746'],
 'dp_hist_alt_bin_freq': ['435|2027|3926|7179|10486|12410|11638|10190|8325|6502|4784|3366|2197|1440|1040|722|505|401|275|266'],
 'dp_hist_alt_n_larger': [736],
 'dp_hist_all_bin_freq': ['2789|5201|8107|12604|18963|20827|23056|32220|16192|8

In [382]:
a.INFO['nhomalt']

[21552]

In [383]:
a.ID

'rs17568'

In [384]:
a.INFO['nhomalt_sas']

[4951]

la base _gnomad la arme con vcfanno en base a la distribucion de gnomad

In [413]:

vcf_reader = vcf.Reader( filename='clinvar_20190123_gnomad.vcf',encoding='UTF-8')

for i,record in enumerate(vcf_reader):
    a=record
    if i>20:
        if 'gnomad_homologus_count' in record.INFO.keys():
            break



In [414]:
a.INFO

{'ALLELEID': 556541,
 'CLNDISDB': ['MedGen:C3808739', 'OMIM:615120'],
 'CLNDN': ['Myasthenic_syndrome', '_congenital', '_8'],
 'CLNHGVS': ['NC_000001.10:g.955724G>A'],
 'CLNREVSTAT': ['criteria_provided', '_single_submitter'],
 'CLNSIG': ['Uncertain_significance'],
 'CLNVC': 'single_nucleotide_variant',
 'CLNVCSO': 'SO:0001483',
 'GENEINFO': 'AGRN:375790',
 'MC': ['SO:0001583|missense_variant'],
 'ORIGIN': ['1'],
 'gnomad_homologus_count': [0],
 'gnomad_controls_homologus_count': [0]}

In [2]:
vcf_reader = vcf.Reader( filename='clinvar_20190123_gnomad.vcf',encoding='UTF-8')

clndb_id=[]
clnhom= []

for record in vcf_reader:
    s="0"
    if 'gnomad_controls_homologus_count' in record.INFO.keys():
        s=record.INFO['gnomad_controls_homologus_count'][0]

        clnhom.append(int(s))
        clndb_id.append(int(record.ID))










In [3]:
df_cln_db=pd.DataFrame({"clinvar_id":clndb_id,"homologous_count":clnhom})

In [4]:
rare_variants_dbfnsp=pd.read_csv("rare_control_variants_dbNSFP4.0b2a.txt.tsv",sep="\t")

/home/felipe/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,7,9,11,18,20,25,26,30,33,35,36,38,39,41,44,46,48,49,50,51,53,54,56,57,60,74,76,77,78,79,81,85,86,88,89,90,91,92,93,94,106,219,292,358,359,360,361,362,363,364,365,366) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [5]:
rare_variants_dbfnsp=pd.merge(rare_variants_dbfnsp,df_cln_db,how="left",on=["clinvar_id"])

In [6]:
rare_variants_dbfnsp.to_csv("rare_control_variants_hom_dbNSFP4.0b2a.txt.tsv",sep="\t",index=False)

In [121]:
cln_put_rar_patho=rare_variants_dbfnsp["clinvar_id"].loc[rare_variants_dbfnsp["clinvar_clnsig"].isin(["Pathogenic","Likely_pathogenic",'Pathogenic/Likely_pathogenic','Conflicting_interpretations_of_pathogenicity'])]

In [122]:
len(cln_put_rar_patho)

17818

In [124]:
sum((df_cln_info_pat["clinvar_id"].loc[df_cln_info_pat["has_any_classification"]==True]).isin(cln_put_rar_patho))

13757

nos falta datos de 4000 posiblemente patogenicas raras

In [143]:
df_cln_info_pat.columns

Index(['clinvar_id', 'patho_status', 'auto_dominant', 'auto_recessive',
       'x_recessive', 'x_dominant', 'has_any_classification', 'patho_rev',
       'num_rev'],
      dtype='object')

In [148]:
df_cln_info_rev_vec.loc[df_cln_info_rev_vec["num_rev"]==3]["patho_rev"].iloc[0]

['criteria_provided', '_multiple_submitters', '_no_conflicts']